lib/Partitioner/Partitioner.cpp

```diff
@@ -378,11 +411,38 @@ llvm::Expected<DAGListTy> Partitioner::createDAGWithoutPartition(
   return std::move(partitions);
 }
 
-llvm::Error Partitioner::loadBalancedPartitioning(Function *F,
-                                                  DeviceIDTy numDevices,
-                                                  uint64_t availableMemory,
-                                                  llvm::StringRef backendName,
-                                                  NodeToFunctionMap &mapping) {
+llvm::Expected<DAGListTy>
+Partitioner::loadBalancedPartition(CompilationContext &cctx,
+                                   size_t numDevices) {
+  RETURN_ERR_IF_NOT(
+      module_->getFunctions().size() == 1,
+      "Invalid number of functions in a module. For load-balanced partition "
+      "flow, the module can only contain 1 function");
+  if (multiBackendNames_) {
+    VLOG(1) << "For multi backend types, load-balanced partition can't be "
+               "applied. Call heterogeneous partition instead.";
+    return heterogeneousPartition(cctx);
+  }
+  F_ = selectRepFunc(module_, memSize_);
```

Comment:
nit: since it's guaranteed to be a single function now, should we just get function from the module directly and remove selectRepFunc call?

Reply:
"since it's guaranteed to be a single function now"
-- In fact, we still allow multi-functions in a module as long as none of the function needs partition. That is, for each function, we just create DAG for each function. 

"should we just get function from the module directly and remove selectRepFunc call?"
-- For heterogeneous partition, I think selectRepFunc call should be kept due to the above reason. But for other partition flows, "selectRepFunc" is not necessary.